src/averageAges.js

```diff
@@ -55,7 +75,34 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const requiredWomen = onlyWithSon
+    ? women.filter(woman => people.find(person =>
+      person.mother === woman.name && person.sex === 'm'
```

Comment:
```suggestion
      person.mother === woman.name
```
do you really need this check?